src/app/shared/pipes/public-link.pipe.spec.ts

```diff
@@ -32,10 +33,33 @@ describe('PublicLinkPipe', () => {
         }),
       ],
       folder_linkId: 1001,
-      archiveNbr: '0001-00gp'
+      archiveNbr: '0001-00gp',
     });
     const route = pipe.transform(record);
     expect(route).toBeDefined();
-    expect(route.endsWith('/p/archive/0001-0000/0001-meow/1234/record/0001-00gp')).toBeTruthy();
+    expect(
+      route.endsWith('/p/archive/0001-0000/0001-meow/1234/record/0001-00gp')
+    ).toBeTruthy();
+  });
+
+  it('adds only one slash in the url before p/archive', () => {
```

Comment:
I think this test might be a bit too specific; what we probably  want is to make sure that there are no duplicate slashes in general. Instead of testing in one place (before the `p/archive`), I think it might make more sense to test the entire URL for double slashes (or rather the pathname of the URL since the `https://` bit at the beginning contains double slashes).

To make this testing easy we can take the string returned from `pipe.transform()` and put it into a URL object and test the pathname:
```ts
const url = new URL(pipe.transform(folder));
expect(url.pathname).not.toContain("//");
```